frontend/packages/dev-console/src/components/__tests__/ProjectsExistWrapper.spec.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { FirehoseResult, LoadingBox } from '@console/internal/components/utils';
+import ProjectsExistWrapper from '../ProjectsExistWrapper';
+
+let projects: FirehoseResult;
+
+describe('ProjectsExistWrapper', () => {
+  beforeEach(() => {
+    projects = {
+      data: [],
+      loaded: true,
+      loadError: '',
+    };
+  });
+
+  it('should show loading box', () => {
+    projects.loaded = false;
+    const wrapper = shallow(
+      <ProjectsExistWrapper title="Topology" projects={projects}>
+        {}
+      </ProjectsExistWrapper>,
+    );
+    expect(wrapper.contains(<LoadingBox />)).toBeTruthy();
+  });
+
+  it('should show empty state hint block', () => {
+    const wrapper = shallow(
+      <ProjectsExistWrapper title="Topology" projects={projects}>
+        {}
+      </ProjectsExistWrapper>,
+    );
+    const hintBlock = shallow(wrapper.props().hintBlock);
```

Comment:
This is not a useful test as you are testing another component other than the ProjectExistsWrapper and it is also not a conditional prop value.

Reply:
I just wanted to test the empty state here. If we have no projects then it should render the empty state.

Reply:
If there are no projects assert presence of ODCEmptyState.

Avoid asserting the markup of a component not owned by the component you are testing. Your unit test becomes dependent on the implementation details of another component which is bad.

Reply:
Fixed this. Now, checking for only ```ODCEmptyState``` is present if no project it present.